service/telemetry/config.go

```diff
@@ -30,13 +30,13 @@ var disableAddressFieldForInternalTelemetryFeatureGate = featuregate.GlobalRegis
 type Config struct {
 	Logs    LogsConfig    `mapstructure:"logs"`
 	Metrics MetricsConfig `mapstructure:"metrics"`
```

Comment:
Do these need `omitempty` too?

Reply:
I probably should have included this in the PR description, my bad. I only added `omitempty` to fields where the zero value is also the default. Otherwise the zero value won't be emitted during serialization and it would be impossible to generate configs that set the zero value.

For these fields, we have default configurations for both the `Logs` and `Metrics` fields, so I didn't add `omitempty` so they can be unset if need be.